docs/querying/functions.md

```diff
@@ -464,6 +464,89 @@ by the number of seconds under the specified time range window, and should be
 used primarily for human readability.  Use `rate` in recording rules so that
 increases are tracked consistently on a per-second basis.
 
+## `info()` (experimental)
+
+_The `info` function is introduced as an experiment where we try to improve UX
+around including labels from [info metrics](https://grafana.com/blog/2021/08/04/how-to-use-promql-joins-for-more-effective-queries-of-prometheus-metrics-at-scale/#info-metrics).
+The behavior of this function may change in future versions of Prometheus,
+including its removal from PromQL. info has to be enabled via the 
+[feature flag](../feature_flags.md#experimental-promql-functions) `--enable-feature=promql-experimental-functions`._
+
+`info(v instant-vector, [data-label-selector string])` finds, for each time 
+series in v, all info series with corresponding _identifying_ labels (more on
+this later), and adds the union of their _data_ (i.e., non-identifying) labels
+to the time series. The second, optional, argument is a collection of data
+label matchers that may be used to constrain which info series to consider
+and which data labels to pick from them.
+
+What we mean by identifying labels in this context is the subset that 
+identifies an info metric. The remaining info metric labels are considered
+"data" labels (i.e. non-identifying). The purpose of an info metric's 
+identifying labels is to connect it to regular (non-info) metrics. It's similar
+to foreign keys in SQL, that allow you to combine data spread across multiple
+tables. The data labels in practice encode metadata key value pairs, and are
+what one wants to add to query results (identifying labels are already 
+included).
```

Comment:
I would not go too far astray and pull SQL into the game. Instead, let's better explain the semantics of info series identity.

Suggestion:

```suggestion
Identifying labels of an info series are the subset of labels that uniquely
identify the info series. The remaining labels are considered
_data labels_ (also called non-identifying). (Note that Prometheus's concept
of time series identity always includes _all_ the labels. For the sake of the `info`
function, we “logically” define info series identity in a different way than
in the conventional Prometheus view.) The identifying labels of an info series
are used to join it to regular (non-info) series, i.e. those series that have
the same labels as the identifying labels of the info series. The data labels, which are
the ones added to the regular series by the `info` function, effectively encode 
metadata key value pairs. (This implies that a change in the data labels 
in the conventional Prometheus view constitutes the end of one info series and
the beginning of a new info series, while the “logical” view of the `info` function is
that the same info series continues to exist, just with different “data”.)
```